http-api/src/v1/projects.rs

```diff
@@ -781,8 +752,8 @@ fn resolve_revisions(
 
             let mut changeset: Option<Changeset> = None;
 
-            if let (Ok(commits), Ok(diff), true) = (
-                radicle_source::commits::<PeerId>(browser, None),
+            if let (commits, Ok(diff), true) = (
+                browser.get().iter().cloned().collect::<Vec<git::Commit>>(),
```

Comment:
Same here, this gets the full history (unless you changed it).

Reply:
Just an idea I wanna throw out there while the surf redesign happens and helps with those performance issues -- perhaps you could model some of this stuff with traits. That way you can define these functions generically and supply your own implementation. Then when surf looks better you just have to change what implements the trait :)

Reply:
Hey @FintanH thanks for your input and help over Discord, I want this PR to be merged soon, so I'll propose to keep from adding this abstraction for another PR, and call `radicle_surf` directly.

Reply:
As talked over we keep fetching for now the entire commit history, but since we are paginating we at least do not return all the commits.